packages/node/src/indexer/fetch.service.ts

```diff
@@ -72,15 +74,18 @@ export class FetchService implements OnApplicationShutdown {
   }
 
   async init(): Promise<void> {
-    const subscribeHeads = () =>
-      this.api.rpc.chain.subscribeFinalizedHeads((head) => {
-        this.latestFinalizedHeight = head.number.toNumber();
-        this.eventEmitter.emit(IndexerEvent.BlockTarget, {
-          height: this.latestFinalizedHeight,
-        });
-      });
-    this.api.on('connected', subscribeHeads);
-    await subscribeHeads();
+    this.api.on('connected', () => this.getFinalizedBlockHead());
+    await this.getFinalizedBlockHead();
+  }
+
+  @Interval(FINALIZED_BLOCK_TIME_VARIANCE * 1000)
+  async getFinalizedBlockHead() {
+    const finalizedHead = await this.api.rpc.chain.getFinalizedHead();
```

Comment:
do we need to have some error handling code here?
Not sure what nest will do if error thrown here